packages/peregrine/lib/talons/CartPage/ProductListing/useProduct.js

```diff
@@ -28,17 +28,23 @@ export const useProduct = props => {
     }, []);
 
     const handleRemoveFromCart = useCallback(async () => {
-        setIsUpdating(true);
-        const { error } = await removeItem({
-            variables: {
-                cartId,
-                itemId: item.id
-            }
-        });
+        try {
+            setIsUpdating(true);
+            const { error } = await removeItem({
+                variables: {
+                    cartId,
+                    itemId: item.id
+                }
+            });
 
-        if (error) {
+            if (error) {
+                throw error;
```

Comment:
This change seems unrelated - does decreasing the quantity to zero hit this callback?

Also, what does the UI do with this error, anything?

I was expecting to see L15:

```js
const [removeItem, { data, loading, error }] = useMutation(removeItemMutation);
```

and then the talon returns `data` and `error` back to the UI.

I can understand `await`ing the mutation so we can `setIsUpdating` but our handling here isn't consistent with the rest of the app.

This can also be out of scope of this ticket.

Reply:
Here I opted to mimic the `handleUpdateItemQuantity` callback. I assumed when we worked on handling errors in the UI we could revisit how this is implemented.

Reply:
It's at least better than what we have today, where the UI can get "stuck" in the `isUpdating` state. I'm cool for revisiting.